isis/tests/FunctionalTestsHicrop.cpp

```diff
@@ -0,0 +1,437 @@
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+
+#include "hicrop.h"
+
+#include "TestUtilities.h"
+#include "Fixtures.h"
+#include "gmock/gmock.h"
+
+using namespace Isis;
+using namespace testing; 
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/hicrop.xml").expanded();
+
+TEST_F(MroCube, FunctionalTestHicropCropByCk) {
+  setInstrument("-74999", "HIRISE", "MARS RECONNAISSANCE ORBITER");
+  
+  // make the image stretch outside of the CK time ranges 
+  PvlGroup &inst = testCube->label()->findObject("IsisCube").findGroup("Instrument");
+  inst.findKeyword("DeltaLineTimerCount").setValue("99999");   
+
+  QTemporaryDir prefix;
+  QString outCubeFileName = prefix.path() + "/outTemp.cub";
+  
+  QVector<QString> args = {"from=eh", "to="+outCubeFileName, "source=CK", "CK="+ckPath,
+  "SCLK="+sclkPath, "LSK="+lskPath};
+  
+  UserInterface options(APP_XML, args);
+  Pvl *logs = new Pvl();
+  try {
+     hicrop(testCube, options, logs);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+
+  PvlGroup results = logs->findGroup("Results");
+  
+  EXPECT_EQ((int)results.findKeyword("InputLines"), 1056);
+  EXPECT_EQ((int)results.findKeyword("NumberOfLinesCropped"), 54);
+  EXPECT_EQ((int)results.findKeyword("OututStartingLine"), 55);
+  EXPECT_EQ((int)results.findKeyword("OututEndingLine"), 1056);  
+  EXPECT_EQ((int)results.findKeyword("OututLineCount"), 1002); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartTime"), "2008-05-17T09:37:24.7300819");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopTime"), "2008-05-17T09:37:31.0666673");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartClock"), "4/0895484265.14186"); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopClock"), "4/0895484271.36245"); 
+ 
+  Cube oCube(outCubeFileName); 
```

Comment:
Should we encase the cube instantiation in a try/catch?

Reply:
Hm, probably, in case it failed to open. But I think the exception that ISIS throws naturally does the jobn of letting it fail let you know it failed to open the cube.